apps/rules/rule_handlers.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2022 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from typing import Dict
+import logging
+
+from eve.utils import config
+from superdesk import get_resource_service
+from superdesk.metadata.item import CONTENT_STATE, ITEM_TYPE, CONTENT_TYPE, MEDIA_TYPES
+from superdesk.errors import AlreadyExistsError
+
+logger = logging.getLogger(__name__)
+
+
+class RoutingRuleHandler:
+    NAME: str
+
+    def can_handle(self, rule, ingest_item, routing_scheme) -> bool:
+        raise NotImplementedError()
+
+    def apply_rule(self, rule, ingest_item, routing_scheme):
+        raise NotImplementedError()
+
+
+registered_routing_rule_handlers: Dict[str, RoutingRuleHandler] = {}
+
+
+def register_routing_rule_handler(routing_handler: RoutingRuleHandler):
+    if registered_routing_rule_handlers.get(routing_handler.NAME):
+        raise AlreadyExistsError(f"Ingest Publisher: {routing_handler.NAME} already registered")
```

Comment:
imo we're loosing ability to override some core behaviour for client if needed, not sure that's needed

Reply:
Understood, I'll remove this tomorrow